### PR TITLE
Varying Q

### PR DESCRIPTION
**Pull request by rafaelorozco**
Added some parameters to model structures and constructors so that user can interface a spatially varying Q with Q implementation that Mathias has.

### CODE REVIEW

File: src/TimeModeling/ModelStructure.jl [hunk 1]
```diff
@@ -51,6 +51,8 @@ mutable struct Model
     nb::Integer # number of absorbing boundaries points on each side
     m   # slowness squared
     rho # density in g / m^3 (1 for water)
+    Q   # quality factor 
+    f0  # reference frequency for quality factor
```
Comment (mloubout) [inline src/TimeModeling/ModelStructure.jl:55]: I still don't like this `f0` floating around everywhere, I'd rather just have `Q/f0` directly, that's cluter that makes things a pain to maintain.

File: src/TimeModeling/auxiliaryFunctions.jl [hunk 2]
```diff
@@ -36,10 +36,17 @@ end
...
     # Set up Python model structure
-    vp = pad_array(sqrt.(1f0./model.m), pad_sizes(model, options))
+    vp  = pad_array(sqrt.(1f0./model.m), pad_sizes(model, options))
     rho = pad_array(model.rho, pad_sizes(model, options))
+    
+    if isnothing(model.Q)
+        Q = nothing
+    else
+        Q = pad_array(model.Q, pad_sizes(model, options))
+    end
```
Comment (mloubout) [inline src/TimeModeling/auxiliaryFunctions.jl:46]: `Q = isnothing(model.Q) ? nothing : pad_array(model.Q, pad_sizes(model, options))`

File: src/pysource/models.py [hunk 3]
```diff
@@ -114,14 +120,20 @@ def __init__(self, origin, spacing, shape, space_order, nbl=20,
...
+            self._Q_function = np.log(100)
+        else:
+            self._Q_function = self._gen_phys_param(Q, 'Q', space_order)
+
         if self.nbl != 0:
             # Create dampening field as symbol `damp`
             self.damp = Function(name="damp", grid=self.grid)
-            initialize_damp(self.damp, self.padsizes, spacing, fs=fs)
-            self._physical_parameters = ['damp']
+            initialize_damp(self.damp, self._Q_function, f0, self.padsizes, spacing, fs=fs)
```
Comment (mloubout) [inline src/pysource/models.py]: That's quite convoluted and force the creation of an extra function `Q` and of an unused `_physical_parameters`, `initialize_damp(self.damp, Q, ...)` should be enough

File: src/TimeModeling/ModelStructure.jl (hunk 1 above)
Comment (rafaelorozco) [reply src/TimeModeling/ModelStructure.jl:55]: Okay I will make this change

File: src/TimeModeling/auxiliaryFunctions.jl (hunk 2 above)
Comment (rafaelorozco) [reply src/TimeModeling/auxiliaryFunctions.jl:46]: Sure thing!

**Comment by mloubout** [comment]
ping

**Pull request closed without merging**